FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/MainSignInFragment.kt

```diff
@@ -196,11 +214,17 @@ class MainSignInFragment : BaseSingInFragment() {
         val account = googleSignInAccount?.account?.name ?: return
         val idToken = googleSignInAccount?.idToken ?: return
         uuid = SecurityUtils.generateRandomUUID()
-        if (JavaEmailConstants.EMAIL_PROVIDER_GMAIL.equals(EmailUtil.getDomain(account), true)) {
+
+        val regularDomains = arrayOf(
+          JavaEmailConstants.EMAIL_PROVIDER_GMAIL,
+          JavaEmailConstants.EMAIL_PROVIDER_GOOGLEMAIL
+        )
+
+        if (EmailUtil.getDomain(account).toLowerCase(Locale.US) in regularDomains) {
           domainRules = emptyList()
           onSignSuccess(googleSignInAccount)
         } else {
-          uuid?.let { enterpriseDomainRulesViewModel.getDomainRules(account, it, idToken) }
+          uuid?.let { ekmLoginViewModel.fetchPrvKeys(account, it, idToken) }
         }
```

Comment:
This code looks like all non-regular domains (like `hello.com`) will stop working unless they install EKM on their domain. Maybe I'm seeing this wrongly

Reply:
This code runs only when we use `Gmail Sing-in option`. Then if a domain not in `[googlemail.com, gmail.com]` we do the following:

1. login
2. get domain rules

If one of those steps will be failed a user with `hello.com` will no be able to continue. Also, it will be true if `domainRulesResult.data?.orgRules == null`(if a server doesn't return `domain_org_rules` object in a response)

What about the following questions for non-registered domains? (server-side):

- Will `login` be successful for `hello.com`?
- Will `get domain rules` be successful for `hello.com`?
- Do we return a non-null object `domain_org_rules` for `hello.com`?

Reply:
> If one of those steps will be failed a user with hello.com will no be able to continue. Also, it will be true if domainRulesResult.data?.orgRules == null(if a server doesn't return domain_org_rules object in a response)

Not letting user continue if there is error is correct.

> Will login be successful for hello.com?

definitely

> Will get domain rules be successful for hello.com?

definitely

> Do we return a non-null object domain_org_rules for hello.com?

definitely. We return `domain_org_rules: {flags: []}`

This flow needs to remain working for users on `hello.com`, as it's implemented in the app right now it seems like it would not work.